.github/workflows/test.yaml

```diff
@@ -253,17 +261,45 @@ jobs:
           # Stop servers
           pkill kitsune2-bootst
 
+  archive_bundles:
```

Comment:
Split into a separate workflow. You can think of this as:

- Everything useful from the previous step was cached on Cachix.
- All the Rust caching and Nix store that was build in the last job is dropped by switching machine.
- We can download anything that's useful for this job from Cachix.

This actually has to download a bunch of software that the previous steps haven't needed too, so it seemed best to have it be separate.

I have kept a `needs:` here because we don't want to bundle the scenarios and archive them if they're not working!